src/aind_metadata_mapper/utils.py

```diff
@@ -90,15 +91,21 @@ def metadata_service_helper(url: str, timeout: int = 60) -> Optional[dict]:
     Optional[dict]
         Metadata as a dictionary, or None if error occurs.
     """
-    try:
-        response = requests.get(url, timeout=timeout)
-        if response.status_code == 400:
+    max_attempts = 3
```

Comment:
It might be better to add the retry logic in a separate PR. There are third-party libraries that are better for retrying requests session objects.